### PR TITLE
CNV Metrics for WGS

### PR DESCRIPTION
**Pull request by SophieS9**
Addition of CNV cut offs for WGS data. 
- Average depth > 30X (pass/fail) and total CNV count < 300 (warning only) (See accompanying validation summary). 
- Will display for all runs going forward, but we won't pass/fail on this metric until SOP changes are implemented.
- Unit tests added and all run successfully.

### CODE REVIEW

File: config/config_gen01.yaml
Comment (Istaisa) [inline config/config_gen01.yaml:1]: would be useful to add the new metrics to the local config too?

File: pipelines/dragen_pipelines.py [hunk 1]
```diff
@@ -576,4 +576,30 @@ def get_ploidy_metrics(self):
...
+		results_path = Path(self.results_dir)
+		
+		dragen_cnv_metrics_dict = {}
+		
+		for sample in self.sample_names:
+		
+			cnv_metrics_file = results_path.joinpath(sample).glob(f'*{sample}.cnv_metrics.csv')
+			
+			
+			if len(list(cnv_metrics_file)) == 1:
```
Comment (Istaisa) [inline pipelines/dragen_pipelines.py:595]: might be better as a try/except to catch silent errors (I'm assuming that the list length should always be 1)

File: qc_database/templates/auto_qc/view_run_analysis.html [hunk 2]
```diff
@@ -702,6 +729,22 @@ <h2> Sample Level QC </h2>
...
+			{% if sample_analysis.passes_average_coverage %}
+			<td class="table-success">{{ sample_analysis.get_average_coverage }}</td>
+			{% else %}
+			<td class="table-danger">{{ sample_analysis.get_average_coverage }}</td>
+			{% endif %}
+			
+			{% if sample_analysis.passes_cnv_count %}
+			<td class="table-success">{{ sample_analysis.get_cnv_count }}</td>
+			{% else %}
+			<td class="table-warn">{{ sample_analysis.get_cnv_count }}</td>
```
Comment (Istaisa) [inline qc_database/templates/auto_qc/view_run_analysis.html:744]: table-warning not table-warn

**Review by Istaisa — CHANGES_REQUESTED** [review]
All unit tests pass. Tried with test data, also edited to remove results folder and to get a CNV number warning. All looks good! A couple of minor suggestions but otherwise happy to merge.

**Comment by SophieS9** [comment]
Code review changes
- Added parameters to local config - https://github.com/AWGL/auto_qc/pull/125/commits/35fe97d61d93d1a811cc9ffd3eec49f7ab2d6608 
- For the second request, I've added in an else to return an empty dictionary. https://github.com/AWGL/auto_qc/pull/125/commits/20fc497230e473476e0c4f067272b481008371f4
- Added a couple comments to the models - https://github.com/AWGL/auto_qc/pull/125/commits/49e21e216500697c8981bcb3b14575f7c810d151
- Fixed typo on templates - https://github.com/AWGL/auto_qc/pull/125/commits/497fef709731c137684c4f517d95aff02a4f6c56

Checked upload and still working. 
Unit tests still work after changes.

**Review by Istaisa — APPROVED** [review]
Changes look good - ready to merge to development.

**Pull request merged**